x/market/types/types.go

```diff
@@ -85,10 +89,24 @@ func (o Order) ValidateCanMatch(height int64) error {
 		return err
 	}
 
-	if o.StartAt > height {
-		return errors.Errorf("too early to match order (%v > %v)", o.StartAt, height)
+	switch {
+	case height < o.StartAt:
+		return errors.Wrap(ErrOrderTooEarly, fmt.Sprintf("(%v > %v)", o.StartAt, height))
+	case height > o.CloseAt:
+		if o.CloseAt == int64(0) { // zero-value CloseAt value; invalid if unset
```

Comment:
don't worry about backwards-compatibility right now.

please remember to return early for simple cases.

Reply:
removed compatibility case, short circuited normal case